src/components/portal/portal.tsx

```diff
@@ -52,31 +52,45 @@ export const INSERT_POSITIONS: OuiPortalInsertPosition[] = keysOf(
 );
 
 type OuiPortalInsertPosition = keyof typeof insertPositions;
+export type OuiPortalInsert = ExclusiveUnion<
+  { root?: HTMLElement },
+  { sibling: HTMLElement; position: OuiPortalInsertPosition }
+>;
 
 export interface OuiPortalProps {
   /**
    * ReactNode to render as this component's content
    */
   children: ReactNode;
-  insert?: { sibling: HTMLElement; position: OuiPortalInsertPosition };
-  portalRef?: (ref: HTMLDivElement | null) => void;
```

Comment:
This expands what `ref` an be, not contracts, correct? I.e. with this change, `ref` can be a larger set of inputs, rather than a smaller?

Reply:
Yeap, it expands